arch/xtensa/core/userspace.S

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2022, Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <xtensa-asm2-s.h>
+#include <offsets.h>
+#include <offsets_short.h>
+#include <zephyr/syscall.h>
+#include <zsr.h>
+
+/**
+ *  syscall number     arg1, arg2, arg3, arg4, arg5, arg6
+ *  --------------     ----------------------------------
+ *  a2                 a6,   a3,   a4,   a5,   a8,   a9
+ *
+ **/
+.pushsection .text.z_xtensa_do_syscall, "ax"
+.global	z_xtensa_do_syscall
+.align	4
+z_xtensa_do_syscall:
```

Comment:
There are some complications in using the same handler, first the whole L1 handler is a bunch of macros that have static parameters, that are not the same for syscall. The L1 handler is composed by an entry code (defined by the macro `EXCINT_HANDLER`) and a static C handler.  `EXCIANDLER` (entry code)  is a macro with static parameter to basically tell which stack to use in the exception.
All exceptions/interruptions  switch stack, from the current thread stack to the interruption stack, while on syscalls we have to switch to the current's thread privileged stack, also handling the syscall in a C handler can be quite complicated because of syscall parameters and all callN calls that happened until we get into the C code.

So while it is doable, it needs reasonably changes in these macros and handlers. It is not sure for me that this would make the code cleaner, honestly navigate through all these macro is already confuse enough IMHO.